HDDS-10112. Dashboard for read key metrics

muskan1012 opened the pull request:
## What changes were proposed in this pull request?

Created a new dashboard and added visualisations for all the required ReadKey Metrics.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10112

## How was this patch tested?

The patch has been tested over a Grafana.
![image](https://github.com/apache/ozone/assets/55579937/66ef2a05-a90d-4384-bb10-6890433ea9f2)

tanvipenumudy reviewed (commented):
Thank you @muskan1012, for working on the patch. 

Could you please share snapshots to the added dashboards over the PR?
Here are a few references: [Dashboard for Chunks & Volume IO](https://github.com/apache/ozone/pull/6554), [S3 API-level Dashboard](https://github.com/apache/ozone/pull/6202)

tanvipenumudy commented:
@devabhishekpal could you also please take a look? Thanks!

muskan1012 commented:
Sure @tanvipenumudy, please find snapshots here: 
![image](https://github.com/apache/ozone/assets/55579937/a8f29720-ab5b-41f3-b3ac-0830a4fd0aba)
![image](https://github.com/apache/ozone/assets/55579937/66b8015b-6d32-4746-a60d-baf9acd4a02a)
![image](https://github.com/apache/ozone/assets/55579937/401c30f0-83d7-4218-930e-41127fa69b09)

devabhishekpal reviewed (approved):
This looks good to me.
Thanks @muskan1012 for the change.
+1

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              },
+              "disableTextWrap": false,
+              "editorMode": "code",
+              "expr": "rdb_metrics_num_db_key_get_if_exist_checks{instance=~\"ccycloud-.*.mus-t1.root.comops.site:9875\"}",
```
Please fix the hardcoded host value here.

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              },
+              "disableTextWrap": false,
+              "editorMode": "code",
+              "expr": "rdb_metrics_num_db_key_get_if_exist_gets{instance=~\"ccycloud-.*.mus-t1.root.comops.site:9875\"}",
```
Same as the previous comment, please fix the hardcoded host value here.

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              },
+              "disableTextWrap": false,
+              "editorMode": "builder",
+              "expr": "table_cache_metrics_hit_count{instance=~\"ccycloud-.*.mus-t1.root.comops.site:9875\"}",
```
Same as the previous comment, please fix the hardcoded host value here as well.

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              },
+              "disableTextWrap": false,
+              "editorMode": "code",
+              "expr": "table_cache_metrics_miss_count{instance=~\"ccycloud-.*.mus-t1.root.comops.site:9875\"}",
```
Same as the previous comments.

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              },
+              "disableTextWrap": false,
+              "editorMode": "code",
+              "expr": "ugi_metrics_login_success_avg_time{instance=~\"ccycloud-.*.mus-t1.root.comops.site:9877\"}",
```
Please review this instance as well.

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              },
+              "disableTextWrap": false,
+              "editorMode": "code",
+              "expr": "ugi_metrics_login_success_num_ops{instance=~\"ccycloud-.*.mus-t1.root.comops.site:9877\"}",
```
Same as the previous comments, please review this instance as well.

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              "useBackend": false
+            }
+          ],
+          "title": "Rdb no. of db key metrics",
```
We may keep the heading consistent with the other charts under the same section.
```suggestion
          "title": "Rocksdb no. of db key metrics",
```

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              "useBackend": false
+            }
+          ],
+          "title": "Om num key metrics",
```
```suggestion
          "title": "OM num key metrics",
```

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              "useBackend": false
+            }
+          ],
+          "title": "Om num key ops metrics",
```
```suggestion
          "title": "OM num key ops metrics",
```

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+          "type": "timeseries"
+        }
+      ],
+      "title": "OM Servicelists Metrics",
```
```suggestion
      "title": "OM Service Lists Metrics",
```

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+          "type": "timeseries"
+        }
+      ],
+      "title": "OM Read key Info Metrics",
```
```suggestion
      "title": "OM Read Key Info Metrics",
```

tanvipenumudy commented on hadoop-ozone/dist/src/main/compose/common/grafana/dashboards/Ozone - ReadKey Metrics.json:
```diff
@@ -0,0 +1,3664 @@
+              "useBackend": false
+            }
+          ],
+          "title": "Rocksdb level0 metrics (num files )",
```
```suggestion
          "title": "Rocksdb level0 metrics (num files)",
```

tanvipenumudy reviewed (commented):
Please find a few minor nits.

muskan1012 commented:
@tanvipenumudy thanks for the review, I have addressed all of your comments and fixed minor nits as well.

Here I am attaching the Snapshot of the Dashboard with updated json.
![image](https://github.com/apache/ozone/assets/55579937/8a14173c-630a-49e7-a1f3-bf1cd1c532b9)
![image](https://github.com/apache/ozone/assets/55579937/8ff8a5fc-b56f-431d-bff3-9b4ce0f0d96c)
![image](https://github.com/apache/ozone/assets/55579937/9dfc99de-94a1-4d33-b79f-a5d82be458d2)

muskan1012 commented:
Thank you @devabhishekpal for reviewing it.

tanvipenumudy reviewed (approved):
Thank you @muskan1012 for working on the patch, the changes LGTM. Thank you @devabhishekpal for reviewing.
